### PR TITLE
Receive advertisement events on web

### PR DESCRIPTION
**Pull request by rohitsangwan01**
### **PR Type**
Enhancement, Documentation, Bug fix

___

### **Description**
- Improved logging for BLE scan results by replacing `debugPrint` with `log`.
- Added `withManufacturerData` filter to scan filter widget.
- Added error handling for service discovery in peripheral detail page.
- Added `receivesAdvertisements` property to `BleDevice` and method to `UniversalBle`.
- Improved advertisement handling and cleanup on web.
- Added new launch configuration for web development in VSCode.
- Updated changelog for new features and improvements.
- Fixed typos and clarified usage of `withServices` parameter on web in README.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>6 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>home.dart</strong><dd><code>Improve logging for BLE scan results.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

example/lib/home/home.dart

<li>Added <code>dart:developer</code> import.<br> <li> Replaced <code>debugPrint</code> with <code>log</code> for scan results.<br>

</details>

  </td>
  <td><a href="https://github.com/Navideck/universal_ble/pull/63/files#diff-da1b5bdcb6f7b90bf5ad0c250e3af4c06567c23d1ebe6135325b8d30241a3838">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>scan_filter_widget.dart</strong><dd><code>Add manufacturer data filter to scan widget.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

example/lib/home/widgets/scan_filter_widget.dart

- Added `withManufacturerData` filter to scan filter widget.

</details>

  </td>
  <td><a href="https://github.com/Navideck/universal_ble/pull/63/files#diff-2e16cc2ffa0178b5fb2a9624cc78b337dc2a524ec59fe58a2f05fc616b6e94c7">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ble_device.dart</strong><dd><code>Add receivesAdvertisements property to BleDevice.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/src/models/ble_device.dart

<li>Added <code>receivesAdvertisements</code> property to <code>BleDevice</code>.<br> <li> Improved documentation for connection state.<br>

</details>

  </td>
  <td><a href="https://github.com/Navideck/universal_ble/pull/63/files#diff-ee570ae7012a23746736f11e460a03cc3f91a7a215ea081a588a145ac0931247">+8/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>universal_ble.dart</strong><dd><code>Add receivesAdvertisements method to UniversalBle.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/src/universal_ble.dart

<li>Added <code>receivesAdvertisements</code> method.<br> <li> Improved documentation for connection state.<br>

</details>

  </td>
  <td><a href="https://github.com/Navideck/universal_ble/pull/63/files#diff-0299ec0dda75c972b21ca1a2d8ac5fc87bc4dc3f354c49c7995bcb3d18054304">+13/-3</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>universal_ble_platform_interface.dart</strong><dd><code>Add receivesAdvertisements method to platform interface.</code>&nbsp; </dd></summary>
<hr>

lib/src/universal_ble_platform_interface.dart

- Added `receivesAdvertisements` method to platform interface.

</details>

  </td>
  <td><a href="https://github.com/Navideck/universal_ble/pull/63/files#diff-127e48afab502be75a7cc9a9d97749f8a34e499dc343bff386b3da7ddd0ea4b3">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>universal_ble_web.dart</strong><dd><code>Improve advertisement handling on web.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/src/universal_ble_web/universal_ble_web.dart

<li>Added <code>receivesAdvertisements</code> method for web.<br> <li> Improved advertisement watching and cleanup.<br>

</details>

  </td>
  <td><a href="https://github.com/Navideck/universal_ble/pull/63/files#diff-dc72e700f7c098e3f4e71b5d1f38c8514d3007af606d0fd15b1e5f0ecbd580cd">+61/-44</a>&nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Error handling</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>peripheral_detail_page.dart</strong><dd><code>Add error handling for service discovery.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

example/lib/peripheral_details/peripheral_detail_page.dart

<li>Wrapped service discovery in a try-catch block.<br> <li> Added error logging for service discovery.<br>

</details>

  </td>
  <td><a href="https://github.com/Navideck/universal_ble/pull/63/files#diff-d27c53e1f72b2e62946d82111ebe75b0ddf554cabd123d995bd8cc6d91f7d34b">+13/-9</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>launch.json</strong><dd><code>Add launch configuration for web development.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.vscode/launch.json

<li>Added new launch configuration for web.<br> <li> Added task to open Chrome with specific arguments.<br>

</details>

  </td>
  <td><a href="https://github.com/Navideck/universal_ble/pull/63/files#diff-bd5430ee7c51dc892a67b3f2829d1f5b6d223f0fd48b82322cfd45baf9f5e945">+30/-1</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update changelog for new features and improvements.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

CHANGELOG.md

<li>Added entries for receiving advertisement events on web.<br> <li> Added entries for improved cleanup after disconnection on web.<br>

</details>

  </td>
  <td><a href="https://github.com/Navideck/universal_ble/pull/63/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Fix typos and clarify web usage in README.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

README.md

<li>Fixed typos in documentation.<br> <li> Clarified usage of <code>withServices</code> parameter on web.<br>

</details>

  </td>
  <td><a href="https://github.com/Navideck/universal_ble/pull/63/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: lib/src/universal_ble.dart [hunk 1]
```diff
@@ -225,6 +225,13 @@ class UniversalBle {
     );
   }
 
+  /// On Web, it returns true if the web browser supports [watchAdvertisements]. The rest of the platforms will always return true.
+  /// If [canWatchAdvertisements] returns true on `Web`, then you will get scanResult updates of the selected device.
+  /// Not every browser supports this API yet. It is hidden behind the `chrome://flags/#enable-experimental-web-platform-features` flag.
+  /// Even if the device technically has the method, sometimes it won't update ScanResults even though the device may be sending them.
```
Comment (fotiDim) [inline lib/src/universal_ble.dart:231]: This line is unclear

**Review by fotiDim — COMMENTED** [review]
Let's update the changelog and readme with any API changes.

File: lib/src/universal_ble.dart (hunk 1 above)
Comment (rohitsangwan01) [reply lib/src/universal_ble.dart:231]: This is basically for Web, to simply tell us if the Browser can update ScanResults of the selected device ( Device we select from that Web Dialog ), then if the browser supports `watchAdvertisement` then we can wait for the required updates like Manufacturer data or services to be updated by the browser
Note: Most of the browsers need experimental flag for this feature

**Pull request merged**